sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/ShuffleExchangeExec.scala

```diff
@@ -264,7 +264,7 @@ object ShuffleExchangeExec {
             prefixComparator,
             prefixComputer,
             pageSize,
-            canUseRadixSort)
+            false /* canUseRadixSort */)
```

Comment:
Why not use the same check as used in SortExec?

Reply:
We are comparing binary here, so `SortPrefixUtils.canSortFullyWithPrefix` always return false.

Reply:
Ah ok, I did not know that we use binary comparisons. Is this because of Map not being comparable? If it is, then that might be problematic in itself, because you expect the retried stage to return map elements in the same order right?

Reply:
Anyway, back to the code, the change looks good to me. It might help to add a more insightful comment here.

Reply:
> Is this because of Map not being comparable

Yes, and it's also because we don't have to do an expensive normal sort.

The only problem we want to fix here is inputs with random order. Here we just want the inputs to have a stable order, but don't really care what the order is. So comparing with the unsafe row binary format is good enough here.

Reply:
Agree a better comment would be nice here, specifically reference the jira or say don't enable because...

Reply:
Yes, we compare the UnsafeRow binary here, and we the sort here must have a stable result. Otherwise, it will cause the correctness bug.
Thanks for the advice, add more comments done.